core/indigo-core/molecule/src/inchi_wrapper.cpp

```diff
@@ -603,10 +603,11 @@ void InchiWrapper::saveMoleculeIntoInchi(Molecule& mol, Array<char>& inchi)
         }
 
     Molecule* target = &mol;
-    Obj<Molecule> dearom;
+    std::shared_ptr<Molecule> dearom;
+
     if (has_aromatic)
     {
-        dearom.create();
+        dearom = std::shared_ptr<Molecule>();
```

Comment:
Looks like we can use `unique_ptr` here, lifetime of the object is straight and limited. Moreover, we can move declaration of `dearom` inside `if` clause.